### DISCUSSION TITLE
add DVT&FCH

### DISCUSSION THREAD

**Discussion by belmix** [coinfoundry/miningcore#936] [General]
Check and send PR

https://github.com/belmix/miningcore/tree/DVT%26FCH

**Comment by oliverw** [comment]
Yes?
________________________________
From: Bravo ***@***.***>
Sent: Wednesday, September 15, 2021 8:47 AM
To: coinfoundry/miningcore ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [coinfoundry/miningcore] add DVT&FCH (Discussion #936)


Check and send PR

https://github.com/belmix/miningcore/tree/DVT%26FCH<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbelmix%2Fminingcore%2Ftree%2FDVT%2526FCH&data=04%7C01%7C%7Cc45a3f1e60244624d35c08d978256b7a%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637672924373985191%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NCJWEt%2Bbj%2B7v6vN6pNcnnoFSrmA577qEz5Y%2Fy5DX8xM%3D&reserved=0>

**Comment by belmix** [comment]
I need to fix it a bit to make it fit the master branch
DaemonResponse --> RpcResponse
minerName --> context.Miner
workerName --> context.Worker

I'm making amendments, it will take time

**Comment by belmix** [comment]
fixed and sent PR

**Comment by J-A-M-P-S** [comment]
Don´t forget to hide the walletPassword in api output under paymentProcessing extra.

**Reply by belmix** [reply]
> Don´t forget to hide the walletPassword in api output under paymentProcessing extra.

in devault.conf
rpcallowip=127.0.0.1

or separate the api and payment processes

**Reply by J-A-M-P-S** [reply]
devault need a wallet unlock same like ergo

**Reply by J-A-M-P-S** [reply]
i have implented dvt since long and mined tons of blocks

**Reply by belmix** [reply]
Devault is tested and works the same as CustomFee (BlackList) https://github.com/coinfoundry/miningcore/pull/933

**Comment by J-A-M-P-S** [comment]
otherwise on real net you get:
[2021-09-17 17:11:28.9573] [E] [dvt] [Bitcoin Payout Handler] Wallet is locked but walletPassword was not configured. Unable to send funds.
